benchmarks/distributed/lambda_runtimes/cpp_runtime/main.cpp

```diff
@@ -0,0 +1,217 @@
+#include <aws/core/utils/logging/LogMacros.h>
+#include <aws/core/utils/json/JsonSerializer.h>
+#include <aws/core/utils/HashingUtils.h>
+#include <aws/core/platform/Environment.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <aws/core/auth/AWSCredentialsProvider.h>
+
+#include <aws/lambda-runtime/runtime.h>
+
+#include "main.h"
+#include "sighandler.h"
+
+#include <iostream>
+#include <sstream>
+#include <cstdlib>
+
+#include <Python.h>
+#include <JITCompiler.h>
+#include <Utils.h>
+#include <Timer.h>
+#include <VirtualFileSystem.h>
+#include <RuntimeInterface.h>
+
+#include <google/protobuf/util/json_util.h>
+
+using namespace aws::lambda_runtime;
+
+// EXPERIMENT CODE ------------------------
+static uint64_t timestamp() {
+    return std::chrono::duration_cast<std::chrono::nanoseconds>(
+            std::chrono::system_clock::now().time_since_epoch())
+            .count();
+}
+
+std::string res;
+struct Timestamps {
+    uint64_t MAIN_TS;
+    uint64_t READY_TS;
+    uint64_t PYTHON_READY_TS;
+    uint64_t FUNCTION_TS;
+    bool reused;
+
+    void setString(double func_res) {
+        std::stringstream ss;
+        ss << "{"
+           << "\"func_res\":" << func_res
+           << ", \"reused\":" << reused
+           << ", \"body\": {"
+             << "\"MAIN_TS\":" << MAIN_TS
+             << ", \"READY_TS\":" << READY_TS
+             << ", \"PYTHON_READY_TS\":" << PYTHON_READY_TS
+             << ", \"FUNCTION_TS\":" << FUNCTION_TS
+           << "}}";
+        reused = true; // next time, it will be true
+
+        res = ss.str();
+    }
+
+    Timestamps() : MAIN_TS(0), READY_TS(0), PYTHON_READY_TS(0), FUNCTION_TS(0), reused(false) {}
+};
+
+Timestamps timings;
+
+static std::string decodeFunction(const std::string &func_str, int &r) {
+    // do some stupid stuff to get around not having protobuf
+    auto b64Module = PyImport_ImportModule("base64");
+    if(!b64Module) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't open module base64";
+    }
+    auto b64decode = PyObject_GetAttrString(b64Module, "b64decode");
+    if (!b64decode) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't get function base64.b64decode";
+    }
+    auto py_pickled_func_bytes = PyObject_CallFunctionObjArgs(b64decode, PyUnicode_FromString(func_str.c_str()), nullptr);
+    if(!py_pickled_func_bytes) {
+        PyErr_Print();
+        r = -1;
+        return "base64.b64decode failed!";
+    }
+//    std::cout << "pickled func addr: "<< (void*)py_pickled_func_bytes << std::endl;
+    char *mem = PyBytes_AsString(py_pickled_func_bytes);
+    if(!mem) {
+        PyErr_Print();
+        r = -1;
+        return "decoded function wasn't bytes?";
+    }
+    size_t mem_size = PyBytes_Size(py_pickled_func_bytes);
+    std::string res;
+    res.reserve(mem_size);
+    res.assign(mem, mem_size);
+    return res;
+}
+
+const char* handler_worker(const char *payload, int &r) {
```

Comment:
this function has a bit of a weird signature. Usually a return code is returned, and the output written to a pointer. Maybe change to

int handler_worker(const char* payload, char **output) ?

Also please add doxygen strings to what this function does, what return codes mean etc. When using magic constants, better use enum class or define constants using static const int MAGIC_CONST = 100l